api/classes/class-lightweight-api.php

```diff
@@ -0,0 +1,283 @@
+			filter_input( INPUT_POST | INPUT_GET, 'utm_suppression', FILTER_SANITIZE_SPECIAL_CHARS ) :
+			null; // phpcs:ignore
+
+		$this->suppress_newsletter_campaigns                      = ! empty( $_REQUEST['suppress_newsletter_campaigns'] ) ? // phpcs:ignore
+			filter_input( INPUT_POST | INPUT_GET, 'suppress_newsletter_campaigns', FILTER_SANITIZE_SPECIAL_CHARS ) :
+			null; // phpcs:ignore
+		$this->suppress_all_newsletter_campaigns_if_one_dismissed = ! empty( $_REQUEST['suppress_all_newsletter_campaigns_if_one_dismissed'] ) ? // phpcs:ignore
+			filter_input( INPUT_POST | INPUT_GET, 'suppress_all_newsletter_campaigns_if_one_dismissed', FILTER_SANITIZE_SPECIAL_CHARS ) :
+			null; // phpcs:ignore
+		$this->has_newsletter_prompt = ! empty( $_REQUEST['has_newsletter_prompt'] ) ? // phpcs:ignore
+			filter_input( INPUT_POST | INPUT_GET, 'has_newsletter_prompt', FILTER_SANITIZE_SPECIAL_CHARS ) :
+			null; // phpcs:ignore
+
+		$this->referer_url = filter_input( INPUT_SERVER, 'HTTP_REFERER', FILTER_SANITIZE_STRING );
+
+		if ( 'inline' !== $this->placement && 'always' === $this->frequency ) {
+			$this->frequency = 'once';
+		}
+	}
+
+	/**
+	 * Verify referer is valid.
+	 */
+	public function verify_referer() {
+		$http_referer = ! empty( $_SERVER['HTTP_REFERER'] ) ? parse_url( $_SERVER['HTTP_REFERER'] , PHP_URL_HOST ) : null; // phpcs:ignore
+		$valid_referers = [
+			$http_referer,
+			// TODO: Add AMP Cache.
+		];
+		$http_host = ! empty( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : null; // phpcs:ignore
+		return ! empty( $http_referer ) && ! empty( $http_host ) && in_array( strtolower( $http_host ), $valid_referers, true );
+	}
+
+	/**
+	 * Get transient name.
+	 */
+	public function get_transient_name() {
+		return sprintf( '%s-%s-popup', $this->client_id, $this->popup_id );
+	}
+
+	/**
+	 * Get suppression-related transient value
+	 *
+	 * @param string $prefix transient prefix.
+	 */
+	public function get_suppression_data_transient_name( $prefix ) {
+		return sprintf(
+			'%s-%s',
+			$prefix,
+			$this->client_id
+		);
+	}
+
+	/**
+	 * Get suppression-related transient value for newsletter-campaign-suppression, which has the order of client_id and key reversed from the others.
+	 *
+	 * @param string $prefix transient prefix.
+	 */
+	public function legacy_get_suppression_data_transient_name_reversed( $prefix ) {
+		return sprintf(
+			'%s-%s',
+			$this->client_id,
+			$prefix
+		);
+	}
+
+	/**
+	 * Complete the API and print response.
+	 */
+	public function respond() {
+		$this->debug['end_time'] = microtime( true );
+		$this->debug['duration'] = $this->debug['end_time'] - $this->debug['start_time'];
+		if ( defined( 'NEWSPACK_POPUPS_DEBUG' ) && NEWSPACK_POPUPS_DEBUG ) {
+			$this->response['debug'] = $this->debug;
+		}
+		http_response_code( 200 );
+		print json_encode( $this->response ); // phpcs:ignore
+		exit;
+	}
+
+	/**
+	 * Return a 500 code error.
```

Comment:
It's `400` now ;)

Reply:
😞 https://github.com/Automattic/newspack-popups/pull/208/commits/588c6c2367c8f43aa91a3cd1750c529623728b78